src/query.js

```diff
@@ -31,6 +32,70 @@ function toQueryString(obj) {
   return qs;
 }
 
+const findStylePreset = (id, stylePresets) => {
+  const flatPresets = [
+    ...stylePresets.filter(p => p.type !== 'sublist'),
+    ...stylePresets.filter(p => p.type === 'sublist').flatMap(p => p.presets),
+  ];
+
+  return flatPresets.find(preset => preset.id === id);
+};
+
+const findBranchPattern = (map, branchPatterns) => {
+  const branchPattern = branchPatterns.find(p => p.id === map.branchPatternId);
+  if (!branchPattern) return null;
+  return {
+    ...branchPattern,
+    url: createBranchUrl(
+      branchPattern.pattern,
+      map.branch,
+      map.branchPatternStyle
+    ),
+  };
+};
+
+const encodeMaps = (maps, config) => {
+  const mapsToEncode = maps.map(m => {
+    let encodeMap = { id: m.id };
+    let configMap;
+
+    if (m.branchPatternId) {
+      configMap = findBranchPattern(m, config.branchPatterns);
```

Comment:
We need to confirm that the config actually has `branchPatterns` since it's an optional property. Otherwise, without it this can error.

Reply:
Good call!